route/middleware.go

```diff
@@ -37,6 +37,25 @@ func (r *Router) queryTokenChecker(next http.Handler) http.Handler {
 	})
 }
 
+func (r *Router) isKeyAllowed(key string) bool {
```

Comment:
Since this is called on every request, it could be nice to be able to check the key in a map instead of looping through a slice each time.

Reply:
Doing that would require changing the data type of the config value to a map, and writing code that parses the current configuration into it both on initial load and on reload. It's a good idea, but it's something that a) we've already lived with for a long time, b) in practice might even be more expensive because most installations just have `["*"]` as the value, and c) and I'd rather do it as [a separate issue](https://github.com/honeycombio/refinery/issues/673) as part of the config rework.